source/common/quic/envoy_quic_client_connection.cc

```diff
@@ -180,8 +183,8 @@ void EnvoyQuicClientConnection::probeWithNewPort(const quic::QuicSocketAddress&
 
   // The probing socket will have the same host but a different port.
   auto probing_socket =
-      createConnectionSocket(connectionSocket()->connectionInfoProvider().remoteAddress(),
```

Comment:
Is it possible to change probeWithNewPort() to take an Envoy address type? In that case, only SPA probing path will need to create new address instance.

Reply:
Not sure what you mean, we could convert from QUICHE address to Envoy address earlier but it's not clear what that would achieve?

Reply:
That is exactly what I mean, making the caller of probeWithNewPort() provide an Envoy address instance. `connectionSocket()->connectionInfoProvider().remoteAddress()` reuses existing address instance, but quicAddressToEnvoyAddressInstance() creates a new object.